widget/embedded/src/store/slices/data.ts

```diff
@@ -67,8 +81,31 @@ export const createDataSlice: StateCreator<
     if (!options || !options?.type) {
       return blockchainsFromState;
     }
-
     const config = get().config;
+
+    if (options.type === 'custom-token') {
+      const supportedBlockchainsFromConfig = getSupportedBlockchainsFromConfig({
+        config,
+      });
+
+      // Filter blockchains from state based on common types and active custom tokens
+      return blockchainsFromState.filter((blockchain) => {
```

Comment:
This filter seems confusing to me. I thought it should be something like:
```javascript
let supportedBlockchainsForCustomTokens: string[] =
        ACTIVE_BLOCKCHAINS_FOR_CUSTOM_TOKENS;

      /*
       * Supported blockchains can be configured and be limited.
       * In this case, we only keep those active blockchains which exist in config.
       */
      if (supportedBlockchainsFromConfig.length > 0) {
        supportedBlockchainsForCustomTokens =
          supportedBlockchainsFromConfig.filter((blockchain) =>
            ACTIVE_BLOCKCHAINS_FOR_CUSTOM_TOKENS.includes(blockchain)
          );
      }

      return blockchainsFromState.filter((blockchain) =>
        supportedBlockchainsForCustomTokens.includes(blockchain.type)
      );
```

to make this work you need to update `widget/embedded/src/constants/customTokens.ts`:
```
export const ACTIVE_BLOCKCHAINS_FOR_CUSTOM_TOKENS: string[] = ...
```

Reply:
I changed my code to your suggested code
But for this, we have to store the names of the blockchains in ACTIVE_BLOCKCHAINS_FOR_CUSTOM_TOKES instead of typing them.
Did I understand correctly?

Reply:
No. I didn't mean to change `blockchain.type` to `blockchain.name`.  By using my code you will go through a typescript error regarding `ACTIVE_BLOCKCHAINS_FOR_CUSTOM_TOKENS` which should be `TransactionType`. I said to fix this issue you can use: 
```
export const ACTIVE_BLOCKCHAINS_FOR_CUSTOM_TOKENS: string[] = [
TransactionType.EVM,
  TransactionType.SOLANA,
]
```

Reply:
Please revert back the code and only add `: string[]` to the const.

Reply:
I understood. So, I corrected a minor issue in your suggested code to get `ACTIVE_BLOCKCHAINS_FOR_CUSTOM_TOKENS` back to how it was.